cpp/include/cudf/binaryop.hpp

```diff
@@ -213,6 +213,54 @@ int32_t binary_operation_fixed_point_scale(binary_operator op,
 cudf::data_type binary_operation_fixed_point_output_type(binary_operator op,
                                                          cudf::data_type const& lhs,
                                                          cudf::data_type const& rhs);
+namespace binops {
+
+/**
+ * @brief Computes output valid mask for op between a column and a scalar
+ */
+rmm::device_buffer scalar_col_valid_mask_and(
+  column_view const& col,
+  scalar const& s,
+  rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+namespace compiled {
+
+/**
+ * @brief Converts scalar to column_view with single element.
+ *
+ * @param scal    scalar to convert
+ * @param stream  CUDA stream used for device memory operations and kernel launches.
+ * @param mr      Device memory resource used to allocate the returned column's device memory
+ * @return        pair with column_view and column containing any auxiliary data to create
+ * column_view from scalar
+ */
+std::pair<column_view, std::unique_ptr<column>> scalar_to_column_view(
```

Comment:
Looked into this some more. Is there some benefit to this function over `make_column_from_scalar` that I'm not seeing? Columns can be converted to column views, so that's no problem. The only other difference I can see is potentially skipping the copy of the device value into a newly allocated column. To me, that feels like an insufficient justification for the duplicated logic.

@rwlee if you or other reviewers agree with my assessment here, one of us can work on a separate PR to remove that, so it doesn't tangle with this PR's changes.

Reply:
Missed this comment when doing changes yesterday. This function already existed, I was just moving it out of the anonymous namespace. `make_column_from_scalar` with size = 0 can definitely be used instead.